File: typescript/src/core/form-10q.ts
/**
 * Structured view of a 10-Q quarterly report, returned by `Filing.obj()`.
 */

import { TenQItem } from "../parsers/items";
import { PeriodicReport, ReportFiling } from "./periodic-report";

/** 10-Q quarterly report: named sections and XBRL-backed financials. */
export class TenQ extends PeriodicReport {
  static SECTION_ITEMS: Record<string, string> = {
    mda: TenQItem.MANAGEMENT_DISCUSSION_AND_ANALYSIS,
    riskFactors: TenQItem.RISK_FACTORS,
  };

  static async create(filing: ReportFiling): Promise<TenQ> {
    let items: Record<string, string> = {};
    try {
      items = await filing.extractItems();
    } catch {
      items = {};
    }
    return new TenQ(filing, items);
  }
}

File: python/src/sec_edgar_toolkit/core/form_10q.py
"""Structured view of a 10-Q quarterly report, returned by ``Filing.obj()``."""

from __future__ import annotations

from ..parsers.items import TenQItem
from .periodic_report import PeriodicReport


class TenQ(PeriodicReport):
    """10-Q quarterly report: named sections and XBRL-backed financials."""

    SECTION_ITEMS = {
        "mda": TenQItem.MANAGEMENT_DISCUSSION_AND_ANALYSIS,
        "risk_factors": TenQItem.RISK_FACTORS,
    }

    def __repr__(self) -> str:
        return f"TenQ(sections={list(self._items.keys())})"
